### ISSUE TITLE
Reset my User ID does not appear to be optional, causing confusion.

### ISSUE THREAD

**Issue by ModernBalloonie** [Yellow-Dog-Man/Resonite-Issues#820]
### Is your feature request related to a problem? Please describe.

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/82356528/df02ec26-b0ab-404a-8ec2-75155b50a84d)
This "Reset my UserID" doesn't appear to be optional and it causes confusion, especially to newer users.

People have reset their userids before because they thought it was required, but it wasn't, and it breaks everything for them.

### Describe the solution you'd like

an (Optional) added to the end of "Reset my User ID"

### Describe alternatives you've considered

i'm not sure what could be considered an alternative.

**Comment by Frooxius** [comment]
We can add the optional, that's an easy tweak.

However it being scary is on purpose, to make people think twice before interacting with it and read the text, since it's not reversible.

I'm a bit confused on what about it would make people think it's not optional. Specifically the text starts with "Choosing this **option**..."  And at the end also "Unless you have a strong reason to reset your UserID, you SHOULD NOT check this **option**."

When user proceeds with this checked, they get additional warning, asking them if they really REALLY want this option.

It seems odd to me, that with that language people would conclude it's not optional - I feel like there's something odd going on here and some more context would help. I'm wondering, would adding "optional" really help here?

**Comment by ProbablePrime** [comment]
I wanted to split the "reset" step to an additional screen similar to the "Overwrite" one, That might allow for some better formatting but I never got around to it.

Users also confuse this page with being required for migration because the reserved password is on there.

**Comment by TisFoolish** [comment]
It's location is what makes it look like it's not optional.  It has become common design to have checkboxes that you must check in order finish signing up for something appear right at the very end, right before you actually register.  Visually separating it from the rest of the process should make it much more obvious that it's not a necessary part of the process.

**Comment by Frooxius** [comment]
I see, thanks for the context.

That might be a better solution, putting it as a separate phase. 

I do feel that if the text does not convince the user that it's not optional, adding extra word probably wouldn't help as much, so restructuring it a bit could have more of an impact.

**Comment by ProbablePrime** [comment]
I'll take this and make it a separate step. Eventually :D

**Comment by TisFoolish** [comment]
No matter how important it is, you'll never be able to get most regular end users to read long text, ***especially*** if it seems like something they have to click anyways.

It's better that you make it seem optional and *not necessary*.

**Comment by Frooxius** [comment]
I think it's important we try, because at least at that point, it's on the user that they didn't pay attention - we tried to make it seem as important as we could.

One problem here is that some users might actually want this option, but if it seems unimportant and optional, they will skim over it and skip it and miss their only opportunity to choose this option - this can be equally problematic.

**Comment by TisFoolish** [comment]
Making it seem important is quite different from making it seem nonoptional. Visually eparating it will solve the nonoptional issue, but you can do tricks to still make it seem important. Eg separate it into a collapsed by default expandable field then when a user clicks on "register account" brightly double flash the field and have a dialogue box appear above it that says "have you checked out this option?"

Making the text shorter while still covering all the bases will also help reduce the count of people who might click it by accident. It repeats the same information multiple times, which works against your goal if wanting them to read and understand it.

**Comment by Zandario** [comment]
This kind of info in UI/UX Design are usually handled by a confirmation dialog "Are you sure you want to do this?"

The entire content is a bit much, yes, but it is still important to communicate what this will actually do if they accept it.

This generally results in less user confusion.

# TL;DR
- click (optional) checkbox
- triggers confirmation modal

Goal: minimizing information overload while still communicating the risks.

**Comment by MrVoltz** [comment]
I would keep the "novel" there and under it I would add two radio buttons:
- [ ] keep U-MrVoltz (recommended)
- [ ] reset to U-sjdjsowiJdjdj (may break existing content)

The user will then have to select one of them (if they try to submit the form, a red message would appear above/below the radio buttons: "Please choose one option"). Also make the whole line of text active (like <label> in HTML), so users don't have to click the teeny tiny circle.

The users will mentally skip the long "novel" and see the two options, would see the random characters there and if they would ask themselves (why would I want that?), then they can read the wall of text above it. But most will go with the clearly marked recommended option I think.

**Comment by Zandario** [comment]
That article has exactly what I'm recommending... But yeah seems Prime has it handled.

**Comment by Banane9** [comment]
Is this still relevant? All reserved accounts have been unlocked since early November last year.

**Comment by Frooxius** [comment]
Oh yeah, this is not relevant anymore.

Closing this. Thanks!
